lib/fluentReports.js

```diff
@@ -427,7 +427,20 @@
             currentLineHeight: function (val) { //noinspection JSPotentiallyInvalidUsageOfThis
                 return this._PDF.currentLineHeight(val); },
             widthOfString: function (d, o) { //noinspection JSPotentiallyInvalidUsageOfThis
-                return this._PDF.widthOfString(d, o); },
+                var splitString = d.split(' '),
+                    curWord,
+                    wos = 0,
+                    wordLength;
+                for (var i = 0; i < splitString.length; ++i) {
+                    curWord = splitString[i];
+                    if ((i + 1) !== splitString.length) {
+                        curWord += ' ';
+                    }
+                    wordLength = this._PDF.widthOfString(curWord, o);
+                    wos += wordLength;
+                }
+
+                return wos; },
```

Comment:
Why are you doing this; the code you replaced calls the main pdf widthOfString.  Why copy it, when it calls it directly so that in the future if it is changed it will still call it...

Reply:
That change solves a wrapping bug, by bringing FluentReports and PDFKit more closely in line (with the wrapping length-of-string) calculations.  Now that you mention it though, it may make more sense to add an option to `widthOfString()` to allow by-word checking, otherwise defaulting to the base _PDF `widthOfString()` call.  I'm going to add such an option now.

This current change does still call _PDF `widthOfString()` of course, but it calls it on each word rather than on the whole string (which isn't necessarily correct in all instances).  I think that we do need this capability though, because the PDFKit line_wrapper file calls `widthOfString()` per word rather than per string when determining wrapping, which results in slight differences when processing PDFKits font/standard.js `widthOfString()`.

Reply:
The first attempt at adding this was done here - 6e03014ccd9a96b2ad25ca6d77775d9501de4870.  However, this doesn't change the above function, but only the main FluentReports `widthOfString()` function.  The comments indicate that the `widthOfString()` function related to this comment is used to emulate how it will wrap things, so splitting by word should be correct here, at least given the current implementation of PDFKit.

What are your thoughts on this?  PDFKit doesn't use the base `widthOfString()` for the whole string when checking wrapping, but instead checks it word by word as it is going through the wrapping process.  That is why calling the base function alone will not work in this instance.  PDFKit takes baseFunc + word looping, so the FluentReports function would need to do the same baseFunc + word looping in order to get matching results.